### PR TITLE
Not contains fix

### PR DESCRIPTION
**Pull request by MortenBoysen**
This should fix issue https://github.com/dennisdoomen/FluentAssertions/issues/426. I also added more NotContain methods to align Contain and NotContain overloads more.

Being completely new to Git and Github, just let me know if I need to change something

### CODE REVIEW

File: Tests/FluentAssertions.Shared.Specs/CollectionAssertionSpecs.cs [hunk 1]
```diff
@@ -1580,6 +1601,13 @@ public void Should_succeed_when_asserting_collection_does_not_contain_an_item_th
         }
 
         [TestMethod]
+        public void Should_succeed_when_asserting_collection_does_not_contain_any_items_that_is_not_in_the_collection()
```
Comment (dennisdoomen) [inline Tests/FluentAssertions.Shared.Specs/CollectionAssertionSpecs.cs]: Please use the same naming conventions and AAA structure as the other specs.

File: Tests/FluentAssertions.Shared.Specs/GenericDictionaryAssertionSpecs.cs [hunk 2]
```diff
@@ -917,6 +917,17 @@ public void Should_succeed_when_asserting_dictionary_does_not_contain_a_key_that
         }
 
         [TestMethod]
+        public void Should_succeed_when_asserting_dictionary_does_not_contain_multiple_keys_from_the_dictionary()
```
Comment (dennisdoomen) [inline Tests/FluentAssertions.Shared.Specs/GenericDictionaryAssertionSpecs.cs]: Please use the same naming conventions and AAA structure as the other specs.

**Comment by dennisdoomen** [comment]
Nice work! 

I only have some minor comments. Are you willing to fix those? If so, I'll squash the commits during merging after you're done.

File: Tests/FluentAssertions.Shared.Specs/GenericDictionaryAssertionSpecs.cs (hunk 2 above)
Comment (MortenBoysen) [inline Tests/FluentAssertions.Shared.Specs/GenericDictionaryAssertionSpecs.cs]: I actually copied both the structure and name from the existing test for Contain. That is why I thought it was correct. I simply changed _contains_ to _does_not_contain_ in the name.

**Comment by MortenBoysen** [comment]
I can fix them, but if those tests are incorrectly named and structured, then there are several tests for Contain that are also incorrect - and perhaps others as well. Should we change all of them?

BTW: Could you give an example of how you would name these tests? The structure is easy to fix.

**Comment by dennisdoomen** [comment]
Yeah, I know. These are old tests. I fix them when I or others touch them. Wrt the naming, I meant the `When_......_it_should_....` mantra.

**Comment by MortenBoysen** [comment]
Ok. I will have a look and fix them. :-)

**Comment by MortenBoysen** [comment]
I hope this fixes the issues. However, I have no idea if I am doing the right thing in Github to actually push the latest name fixes to you.

**Comment by dennisdoomen** [comment]
Looks good. I'll merge and quash your changes.

**Pull request merged**